plugins/ingest/src/test/java/org/elasticsearch/ingest/GrokPipelineIT.java

```diff
@@ -0,0 +1,112 @@
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+
+public class GrokPipelineIT extends ESIntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return pluginList(IngestPlugin.class);
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> transportClientPlugins() {
+        return nodePlugins();
+    }
+
+    public void test() throws Exception {
+        new PutPipelineRequestBuilder(client(), PutPipelineAction.INSTANCE)
+                .setId("_id")
+                .setSource(jsonBuilder().startObject()
+                        .field("description", "my_pipeline")
+                        .startArray("processors")
+                        .startObject()
+                        .startObject("grok")
+                        .field("field", "field1")
+                        .field("pattern", "%{NUMBER:val:float} %{NUMBER:status:int} <%{WORD:msg}>")
+                        .endObject()
+                        .endObject()
+                        .endArray()
+                        .endObject().bytes())
+                .get();
+        assertBusy(new Runnable() {
+            @Override
+            public void run() {
+                GetPipelineResponse response = new GetPipelineRequestBuilder(client(), GetPipelineAction.INSTANCE)
+                        .setIds("_id")
+                        .get();
+                assertThat(response.isFound(), is(true));
+                assertThat(response.pipelines().get("_id"), notNullValue());
+            }
+        });
+
+        createIndex("test");
+        XContentBuilder updateMappingBuilder = jsonBuilder().startObject().startObject("properties")
+                .startObject("status").field("type", "integer").endObject()
+                .startObject("val").field("type", "float").endObject()
+                .endObject();
+        PutMappingResponse putMappingResponse = client().admin().indices()
+                .preparePutMapping("test").setType("type").setSource(updateMappingBuilder).get();
+        assertAcked(putMappingResponse);
+
+        client().prepareIndex("test", "type", "1").setSource("field1", "123.42 400 <foo>")
+                .putHeader("ingest", "_id")
+                .get();
+
+        assertBusy(new Runnable() {
+            @Override
+            public void run() {
+                Map<String, Object> doc = client().prepareGet("test", "type", "1")
+                        .get().getSourceAsMap();
+                assertThat((double) doc.get("val"), closeTo(123.42, 0.01));
```

Comment:
Debugging why this is flaky.

Value of `doc.get("val")` sometimes equals `123.42` and sometimes equals `123.41999816894531`.